src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Contains methods and attributes relevant to storing and operating
+ * on tasks added by user.
+ * 
+ * @author Siau Wee
+ */
+public class TaskList {
+
+    private ArrayList<Task> addedTasks;
+
+    private Storage storage;
+
+    private Ui ui;
```

Comment:
```suggestion
    private final ArrayList<Task> addedTasks;

    private final Storage storage;

    private final Ui ui;
```

Suggesting the addition of `final` modifier.